highlight/highlight
# Move alert emails to lambda

## Summary
The react emails are being stored and executed by a lambda function. Swap over the current alert emails to use these new emails.

Refactored some model code to allow for use of Lambda without circular imports

## How did you test this change?
1) Create an error alert that notifies an email
2) Place a breakpoint in the lambda function
3) Create an error via the UI
4) Catch and copy the the stringified data in `b`
```
"{\"data\":{\"alertLink\":\"https://localhost:3000/1/alerts/errors/1\",\"errorCount\":1,\"errorEvent\":\"\\\"Uncaught ReferenceError: badVariableAccess is not defined\\\"\",\"errorLink\":\"https://localhost:3000/1/errors/E54koz1VBVnz180BHZLmAEWMfrRD/instances/947?referrer=email\",\"firstError\":true,\"projectName\":\"Highlight App\",\"serviceName\":\"frontend\",\"sessionExcluded\":false,\"sessionLink\":\"https://localhost:3000/1/sessions/OZdlHGod5EuCVEedKln5TMMltN53\"},\"template\":\"error-alert\"}"
```
5) Paste this into the test for the [session-insights-email lambda function](https://us-east-2.console.aws.amazon.com/lambda/home?region=us-east-2#/functions/session-insights-email?tab=testing) under the "body" key
```
{
  "body": "{\"data\":{\"alertLink\":\"https://localhost:3000/1/alerts/errors/1\",\"errorCount\":1,\"errorEvent\":\"\\\"Uncaught ReferenceError: badVariableAccess is not defined\\\"\",\"errorLink\":\"https://localhost:3000/1/errors/E54koz1VBVnz180BHZLmAEWMfrRD/instances/947?referrer=email\",\"firstError\":true,\"projectName\":\"Highlight App\",\"serviceName\":\"frontend\",\"sessionExcluded\":false,\"sessionLink\":\"https://localhost:3000/1/sessions/OZdlHGod5EuCVEedKln5TMMltN53\"},\"template\":\"error-alert\"}"
}
```
5) Copy the contents of the response and parse the contents of the body
6) Save in a file and open in a browser (Note: if styles are missing, check that there are no wearing parsing symbols)

![Screenshot 2023-12-06 at 9 38 14 AM](https://github.com/highlight/highlight/assets/17744174/bd55bf2a-6065-4d44-ba8a-3d0e8f01a1f4)

## Are there any deployment considerations?
Confirm each email is send correctly

## Does this work require review from our design team?
N/A
